tests/cross/src/test/scala/tests/pc/PcReferencesSuite.scala

```diff
@@ -0,0 +1,135 @@
+      original: String,
+      compat: Map[String, String] = Map.empty
+  )(implicit location: Location): Unit =
+    test(name) {
+      val edit = original.replaceAll("(<<|>>)", "")
+      val expected = original.replaceAll("@@", "")
+      val base = original.replaceAll("(<<|>>|@@)", "")
+
+      val (code, offset) = params(edit, "Highlight.scala")
+      val ranges = presentationCompiler
+        .references(
+          PcReferencesRequest(
+            CompilerVirtualFileParams(
+              URI.create("file:/Highlight.scala"),
+              code,
+              EmptyCancelToken
+            ),
+            includeDefinition = false,
+            offsetOrSymbol = JEither.forLeft(offset)
+          )
+        )
+        .get()
+        .asScala
+        .flatMap(_.locations().asScala.map(_.getRange()))
+        .toList
+
+      assertEquals(
+        renderRangesAsString(base, ranges),
+        getExpected(expected, compat, scalaVersion)
+      )
+    }
+
+  check(
+    "implicit-args",
+    """|package example
+       |
+       |class Bar(i: Int)
+       |
+       |object Hello {
+       |  def m(i: Int)(implicit b: Bar) = ???
+       |  val foo = {
+       |    implicit val b@@arr: Bar = new Bar(1)
+       |    m<<>>(3)
+       |  }
+       |}
+       |""".stripMargin,
+    compat = Map("3" -> """|package example
+                           |
+                           |class Bar(i: Int)
+                           |
+                           |object Hello {
+                           |  def m(i: Int)(implicit b: Bar) = ???
+                           |  val foo = {
+                           |    implicit val barr: Bar = new Bar(1)
+                           |    m(3)<<>>
+                           |  }
+                           |}
+                           |""".stripMargin)
+  )
+
+  check(
+    "implicit-args-2",
+    """|package example
+       |
+       |class Bar(i: Int)
+       |class Foo(implicit b: Bar)
+       |
+       |object Hello {
+       |  implicit val b@@arr: Bar = new Bar(1)
+       |  val foo = <<>>new Foo
+       |}
+       |""".stripMargin,
+    compat = Map(
+      "3" -> """|package example
+                |
+                |class Bar(i: Int)
+                |class Foo(implicit b: Bar)
+                |
+                |object Hello {
+                |  implicit val barr: Bar = new Bar(1)
+                |  val foo = new Foo<<>>
+                |}
+                |""".stripMargin
+    )
+  )
+
+  // for Scala 3 the symbol in bar reference is missing (<none>)
+  check(
+    "implicit-args-3".tag(IgnoreScala3),
+    """|package example
+       |
+       |class Bar(i: Int)
+       |class Foo(implicit b: Bar)
+       |
+       |object Hello {
+       |  implicit val b@@arr = new Bar(1)
+       |  for {
+       |    _ <- Some(1)
+       |    foo = <<>>new Foo()
+       |  } yield ()
+       |}
+       |""".stripMargin
+  )
+
+  check(
+    "case-class",
+    """|case class Ma@@in(i: Int)
```

Comment:
Why add these two tests with no usages? Is this on puspose?

Reply:
Initially I added all zero-extent symbols, not only implicits, so it showed some additional synthetic symbols in those tests.